distributed/deploy/local.py

```diff
@@ -145,12 +146,24 @@ def __repr__(self):
     def __await__(self):
         return self._started.__await__()
 
+    def sync(self, func, *args, **kwargs):
+        asynchronous = kwargs.pop('asynchronous', None)
+        if asynchronous or self._asynchronous or getattr(thread_state, 'asynchronous', False):
+            callback_timeout = kwargs.pop('callback_timeout', None)
+            future = func(*args, **kwargs)
+            if callback_timeout is not None:
+                future = gen.with_timeout(timedelta(seconds=callback_timeout),
+                                          future)
+            return future
+        else:
+            return sync(self.loop, func, *args, **kwargs)
```

Comment:
Is this related to the grouped worker issue? (I'm not following)

Reply:
Something was slightly nicer if we switched to using self.sync here (this is a common pattern in the client).  I decided to clean this up as part of this PR.